core/src/utils.ts

```diff
@@ -147,3 +148,7 @@ export const getOraklErrorCode = (e, defaultErrorCode) => {
     return defaultErrorCode
   }
 }
+
+export function isAddressValid(addr: string): boolean {
+  return ethers.utils.isAddress(addr) && addr !== addr.toLowerCase() && addr !== addr.toUpperCase()
```

Comment:
just out of curiosity, what's the reason for checking if the addr is not all lower or upper case letters? wouldn't just checking `ethers.utils.isAddress(addr)` be enough? and this is just a suggestion but would be nice to have that reason as a comment

Reply:
The `ethers.utils.isAddress` function considers all-lowercase and all-uppercase addresses as valid. That is not the desired logic which is to make sure the address contains a checksum.